CMakePresets.json

```diff
@@ -0,0 +1,396 @@
+                "CMAKE_C_COMPILER": "gcc",
+                "CMAKE_CXX_COMPILER": "g++"
+            },
+            "vendor": {
+                "microsoft.com/VisualStudioSettings/CMake/1.0": {
+                    "hostOS": [
+                        "Linux"
+                    ]
+                },
+                "microsoft.com/VisualStudioRemoteSettings/CMake/1.0": {
+                    "sourceDir": "$env{HOME}/.vs/$ms{projectDirName}"
+                }
+            }
+        },
+        {
+            "name": "linux-clang-debug",
+            "displayName": "clang Debug",
+            "description": "Target Linux with the clang compiler, debug build type",
+            "generator": "Ninja",
+            "binaryDir": "${sourceDir}/out/build/${presetName}",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Debug",
+                "CMAKE_INSTALL_PREFIX": "${sourceDir}/out/install/${presetName}",
+                "CMAKE_C_COMPILER": "clang",
+                "CMAKE_CXX_COMPILER": "clang++"
+            },
+            "vendor": {
+                "microsoft.com/VisualStudioSettings/CMake/1.0": {
+                    "hostOS": [
+                        "Linux"
+                    ]
+                },
+                "microsoft.com/VisualStudioRemoteSettings/CMake/1.0": {
+                    "sourceDir": "$env{HOME}/.vs/$ms{projectDirName}"
+                }
+            }
+        },
+        {
+            "name": "linux-clang-release",
+            "displayName": "clang Release",
+            "description": "Target Linux with the clang compiler, release build type",
+            "generator": "Ninja",
+            "binaryDir": "${sourceDir}/out/build/${presetName}",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+                "CMAKE_INSTALL_PREFIX": "${sourceDir}/out/install/${presetName}",
+                "CMAKE_C_COMPILER": "clang",
+                "CMAKE_CXX_COMPILER": "clang++"
+            },
+            "vendor": {
+                "microsoft.com/VisualStudioSettings/CMake/1.0": {
+                    "hostOS": [
+                        "Linux"
+                    ]
+                },
+                "microsoft.com/VisualStudioRemoteSettings/CMake/1.0": {
+                    "sourceDir": "$env{HOME}/.vs/$ms{projectDirName}"
+                }
+            }
+        }
+    ],
+    "buildPresets": [
+        {
+            "name": "default-build-windows-msvc-debug",
+            "displayName": "Default",
+            "description": "Vanilla build",
+            "configurePreset": "windows-msvc-debug"
+        },
+        {
+            "name": "verbose-build-windows-msvc-debug",
+            "displayName": "Verbose",
+            "description": "Passes -v to Ninja",
+            "nativeToolOptions": [
+                "-v"
+            ],
+            "configurePreset": "windows-msvc-debug"
+        },
```

Comment:
I don't think adding a separate entry for --verbose is a good idea. If someone runs into such low-level issues (very rare), they would have an easier time reading `build.ninja` file.

```suggestion
```

Reply:
Yes this is why I also have the "Default" build preset for windows-msvc-debug and all other configure-presets. Default will just run ninja, but if something is fishy the user can get more info by selecting this build-preset (or any other verbose)

Reply:
To reduce the number of presets, we can remove the ones that only add a --verbose option (rare usecase). If someone runs into a problem, they can go into their cmake and modify it manually.

Reply:
who want to have less options? The fact that there is a verbose option does not mean that you have to build it that way. Well I can remove that if you insist but when you set up a new project it might be useful to have it out of the box.

Reply:
I think it is desirable to keep the number of presets manageable. Imaging if someone wants to modify a build preset. Now, they have to do it in two places. For the regular one and the verbose one.

Reply:
Fair point! As I can see default/empty build preset already exist, there is no need to define it 
![image](https://user-images.githubusercontent.com/9009334/145731706-54b4854d-cca7-4093-95d9-73dda4e693f7.png)
Shall I remove the user defined Default build presets too?

Reply:
If that translates to other IDEs too, that's probably a good idea.

Reply:
Visual Studio has the same behavior with the VSCode when it comes to the default build preset. It makes no sense to define a preset if you do not alter the option arguments. You could just not pass any preset in the cli! I guess this is what IDEs does too. So if we are going to keep only one build preset for demonstration reasons I would only keep the verbose since the default can be omitted.  Otherwise, I can erase all the buildPresets and if someone want to pass something special he can search a little bit more.

Reply:
@aminya if you have any useful build option to suggest that a build preset could possibly pass to the ninja it would be great.

Reply:
I prefer that we remove the default presets.

Reply:
OK I made that change. We should be good to go now

Reply:
@aminya is this OK in the latest commit? I think we ready to go if you do not have anything else to propose. The default and verbose build presets were deleted as discussed above.

Reply:
@aminya I tried to re-use configurations between different presets as much as possible, so there is one place to define a single flag. Now I have some common presets and other presets inherit from them. I think this is a good addition to minimize the maintenance cost of the CMakePresets.json file.